v2/yarpcfx/yarpcfx.go

```diff
@@ -0,0 +1,72 @@
+package yarpcfx
+
+import (
+	"go.uber.org/fx"
+	yarpc "go.uber.org/yarpc/v2"
+	"go.uber.org/yarpc/v2/yarpcclient"
+	"go.uber.org/yarpc/v2/yarpcrouter"
+)
+
+const _name = "yarpcfx"
+
+// Module provides YARPC integration for services. The module produces
+// a yarpc.Router and a yarpc.ClientProvider.
+var Module = fx.Options(
+	fx.Provide(NewClientProvider),
+	fx.Provide(NewRouter),
+)
+
+// ClientProviderParams defines the dependencies of this module.
+type ClientProviderParams struct {
+	fx.In
+
+	Clients []yarpc.Client `group:"yarpcfx"`
+}
+
+// ClientProviderResult defines the values produced by this module.
+type ClientProviderResult struct {
+	fx.Out
+
+	Provider yarpc.ClientProvider
+}
+
+// NewClientProvider provides a yarpc.ClientProvider to the Fx application.
+func NewClientProvider(p ClientProviderParams) (ClientProviderResult, error) {
+	provider := yarpcclient.NewProvider()
+	for _, c := range p.Clients {
+		provider.Register(c.Service, c)
```

Comment:
Good catch, I updated the `OutboundConfig` so that we can just handle it now.

Reply:
Not quite.

We want users to be able to specify an outbound name that does not match the service name. For this to work we need to register the client by the outbound name, not the service name.

If, for instance, we had multiple outbound names for a single service (that is using the `service: ...` override), this would stomp over the registration.

We'll need to either thread an additional `Name` field in the `yarpc.Client` or retrieve (name, client) tuples as input to get this working. This may be out of scope for this diff, so feel free to skip for now.

Reply:
Ah ok, I see what you mean now. As discussed offline, let's defer the change for now and see if it becomes necessary as we continue to port in the changes - ideally we won't need to have the `Service` override in the long term.